### PR TITLE
chore: create pull request template and fix issue forms

### LINKED ISSUE #92: Issue and Pull Request Templates
## Problem
I noticed that this repository is missing templates for PRs and Issues. 
## Disadvantage 
This can make it difficult for people who are very new to open source to have difficulty with sharing their contribution ideas and submit them.

## Possible Solution
Create Issue templates for specific issues like Documentation, Bugs, Feature requests, Security, etc.
- See this example
```
name: General Issue
description: This template can be used for any type of issue, including bug reports, feature requests, and questions.
labels: ["general"]

body:
  ---
  **Title:** Describe your issue in a clear and concise title.
  **Description:** Provide a detailed description of your issue, including steps to reproduce it and any relevant error messages.
  **Environment:** Include information about your environment, such as the operating system, browser version, and any relevant software versions.
  **Attachments:** If possible, attach any screenshots, screen recordings, or other files that may help to diagnose the issue.
  ---

```
 For pull request templates, you can add comments on how to link issues that are addressed in the PRs, how to show the solution, and specific steps needed to ensure that contributors are submitting their PRs properly. 
 
Here's an example:

```
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
```
## Benefits
This would help new contributors develop good writing habits when raising an issue and filling out a PR form. For maintainers, these tools would help them review these requests and determine if they align with the project's purpose.

### PR DESCRIPTION
**Pull request by CBID2**
## Description
This PR adds a template for pull requests and revised issue forms

## Benefits 
This will help new contributors effectively raise issues and submit their pull requests to the project. 
## Related Issue
Closes #92 

## Note to Reviewers
Hey @XanderRubio! :) I created the template and it's ready for a review! :)

### CODE REVIEW

**Comment by XanderRubio** [comment]
Looks good with the create pull request template ✅ and additionally when checking the templates what is the issue with:
- bugs
- docs
- feature-request
- security 

The error is stating that the body cannot be empty and this might be related to the issue forms are currently in beta and subject to change. Curious to get your thoughts @CBID2 😀

Screenshot 2023-09-19 at 15 58 00

**Comment by CBID2** [comment]
Thanks @XanderRubio! Ooh good observation. I'll get back to you on that

**Comment by CBID2** [comment]
> Looks good with the create pull request template ✅ and additionally when checking the templates what is the issue with:
> 
> * bugs
> * docs
> * feature-request
> * security
> 
> The error is stating that the body cannot be empty and this might be related to the issue forms are currently in beta and subject to change. Curious to get your thoughts @CBID2 😀
> 
> Screenshot 2023-09-19 at 15 58 00

[According to Github](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms#body-cannot-be-empty-when-issue_body-is-false), this error can be fixed by adding ‘body‘ or  deleting the --- (document separator) between the headers and the body section. What do you think @XanderRubio?

File: .github/ISSUE_TEMPLATE/docs.yml
Comment (XanderRubio) [inline .github/ISSUE_TEMPLATE/docs.yml:1]: docs.yml show up without errors✅
Screenshot 2023-09-19 at 17 09 07

File: .github/ISSUE_TEMPLATE/bugs.yml
Comment (XanderRubio) [inline .github/ISSUE_TEMPLATE/bugs.yml:1]: Currently now seeing this error message:
`YAML syntax error: (): did not find expected '-' indicator while parsing a block collection at line 43 column 10.`
Screenshot 2023-09-19 at 17 09 57

**Comment by XanderRubio** [comment]
> > Looks good with the create pull request template ✅ and additionally when checking the templates what is the issue with:
> > 
> > * bugs
> > * docs
> > * feature-request
> > * security
> > 
> > The error is stating that the body cannot be empty and this might be related to the issue forms are currently in beta and subject to change. Curious to get your thoughts @CBID2 😀
> > Screenshot 2023-09-19 at 15 58 00
> 
> [According to Github](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms#body-cannot-be-empty-when-issue_body-is-false), this error can be fixed by adding ‘body‘ or deleting the --- (document separator) between the headers and the body section. What do you think @XanderRubio?

I am looking into it now, and I see with the docs.yml you have that it works, the value required is a boolean. It seems to be a small formatting issue, which is why the GitHub template doesn't recognize the security, bugs, and feature requests templates. 

Additionally, in the GitHub documentation `Common validation errors when creating issue forms` I see the boolean is used for the `required` key. 
Screenshot 2023-09-19 at 17 27 45

**Comment by CBID2** [comment]
I fixed them all @XanderRubio! :)

File: .github/ISSUE_TEMPLATE/security.yml
Comment (XanderRubio) [inline .github/ISSUE_TEMPLATE/security.yml:1]: Ah, I understand now. You needed to remove the quotes and the values should not be in string format. This is good to know.

**Review by XanderRubio — APPROVED** [review]
I have double-checked and all of your yml files show up without errors to be used in templates. Great work, @CBID2 and once again you have taught me a new feature for when constructing the templates. Thank you!

File: .github/ISSUE_TEMPLATE/security.yml
Comment (CBID2) [reply .github/ISSUE_TEMPLATE/security.yml:1]: Yup!:)

**Comment by CBID2** [comment]
For some odd reason @XanderRubio, it says "Unable to Merge" 🤔

**Comment by XanderRubio** [comment]
This pull request successfully closes out issue #92 and is much appreciated. Congrats @CBID2, for your work being merged, and thank you for your time in debugging the templates.

**Comment by XanderRubio** [comment]
> For some odd reason @XanderRubio, it says "Unable to Merge" 🤔

Hmm, do you think this has to do with the setting for this repo on how you saw the message, "Unable to Merge"?

**Pull request merged**